web/src/ui/composer/MessageComposer.tsx

```diff
@@ -388,6 +394,49 @@ const MessageComposer = () => {
 		}
 		evt.preventDefault()
 	}
+	const resolvePreviews = useCallback((
+		text: string,
+		existingPreviews: Record<string, URLPreviewType | "cleared" | null>,
+	) => {
+		const encrypt = !!room.meta.current.encryption_event
+		const previews: Record<string, URLPreviewType | "cleared" | null> = {}
+		let changed = false
+		const urls = text.matchAll(/\bhttps?:\/\/[^\s/_*]+(?:\/\S*)?\b/gi).toArray()
```

Comment:
note to self: calculate the list before calling this function so that we can immediately clear the URL previews if the URLs were deleted